textfinding_indikator1.py

```diff
@@ -0,0 +1,119 @@
+# this code is for TEXT FINDING - INDIKATOR 1
+# input is txt file from preprocess dokumen lama, nama instansi & judul from dokumen lama & baru
+
+import re
+import preprocess_dokbaru as dokbaru
+import preprocess_doklama as doklama
+from utility import *
+
+
+def txtreader(filename, lv, keyword):
+    # func to search keyword in txt file
+
+    # open txt file
+    file = open(f'cleaned_{filename}.txt', 'r')
+
+    idx = 0
+    result = []
+
+    # read line by line from txt
+    for line in file:
+
+        if (lv == 1):
+            if re.search(keyword[0], line, re.IGNORECASE):
+                result.append([idx, line])
+
+        elif (lv == 2):
+            # cek dgn keyword 'domain'. cont: domain arsitektur proses bisnis, domain data dan informasi, domain arsitektur keamanan, dll
+            for key in list(keyword):
+                if key in keyword:
+
+                    # manfaatin regex untuk searching yg not only kata keyword
+                    reg = f'(?:(domain)\s(arsitektur)?\s?({key}))'
+                    if re.search(reg, line, re.IGNORECASE):
+                        result.append([idx, line])
+                        keyword.remove(key)
+
+        elif (lv == 3):
+            # masih pengecekan keyword yg sama dgn level 2, tapi tanpa 'domain'
+            for key in list(keyword):
+                if key in keyword:
+
+                    # gausah regex krn searching nya langsung kata keyword tanpa ada embel embel 'domain'
+                    if re.search(reg, line, re.IGNORECASE):
+                        result.append([idx, line])
+
+                        # hapus key yg udh ditemuin supaya gausah di search lagi
+                        keyword.remove(key)
+
+        else:  # lv == 4
+            # cek dulu keyword utama level 4 which is arsitektur spbe nasional
+            reg = f'(?:(arsitektur)/s+(spbe)\s+(nasional))'
+
+            # kalau ketemu, lanjut cek keyword [integrasi, reviu, dll]
+            if (re.search(reg, line, re.IGNORECASE)):
+                # check if line contains any keyword from list
+                res = [ele for ele in keyword if (ele in line)]
+                if (res):
+                    result.append([idx, line])
+
+        idx += 1
+
+    file.close()
+    return (result)
+
+
+# if __name__ == '__main__':
+#     pdfparser(sys.argv[1])
+
+def ceklvl(filename):
+    list_final = []
+    text_final = ''
+
+    lvl1 = ["arsitektur SPBE"]
+    res1 = txtreader(filename, 1, lvl1)
+
+    # cek if keyword lvl1 is not found, then return as empty string
+    if (not res1):
+        return text_final
+
+    lvl2 = ["(proses)\s+(bisnis)", "(data)\s+(dan)\s+(informasi)", "(infrastruktur)\s+(spbe)",
+            "(aplikasi)\s+(spbe)", "(keamanan)\s+(spbe)", "(layanan)\s+(spbe)"]
```

Comment:
Please use the (`convert_keywords`) function provided in `utility.py` to improve readability and decrease likelihood of typos and other minor mistakes